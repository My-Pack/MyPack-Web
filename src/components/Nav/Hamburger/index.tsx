import Link from "next/link";
import { useState } from "react";
import styled, { css } from "styled-components";

function Hamburger() {
  const [isOpen, setIsOpen] = useState<boolean>(false);
  const onClickHamburger = () => {
    setIsOpen(!isOpen);
  };

  return (
    <>
      <StyledMenuWrapper onClick={onClickHamburger}>
        <Styledhamburger isOpen={isOpen} />
      </StyledMenuWrapper>
      <StyledMenuListWrapper isOpen={isOpen}>
        <Link href="/designCard">
          <StyledMenuElement>Card Design</StyledMenuElement>
        </Link>
        <Link href="/profile">
          <StyledMenuElement>My Card</StyledMenuElement>
        </Link>
        <Link href="/introduce">
          <StyledMenuElement>Introduce MyPack</StyledMenuElement>
        </Link>

        <Link href="/login">
          <StyledMenuElement>Login</StyledMenuElement>
        </Link>
      </StyledMenuListWrapper>
    </>
  );
}

export default Hamburger;

const StyledMenuElement = styled.div`
  display: flex;
  flex-direction: row;

  align-items: center;
  gap: 0.5rem;
  font-size: 1.5rem;
  font-weight: ${({ theme }) => theme.fontWeight.semibold};

  :hover {
    font-weight: ${({ theme }) => theme.fontWeight.bold};
    background-image: linear-gradient(
      to right,
      ${({ theme }) => theme.color.neonMint},
      ${({ theme }) => theme.color.neonPink}
    );
    background-clip: text;
    -webkit-background-clip: text;
    color: transparent;

    transition: ease-in all 0.25s;
    cursor: pointer;
  }
`;

const StyledMenuListWrapper = styled.div<{ isOpen: boolean }>`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: center;
  padding-right: 2rem;
  gap: 10%;
  width: 25vw;
  height: 100vh;
  background-color: rgb(32, 32, 32, 0.7);
  position: fixed;
  transition: right 0.8s ease;

  right: ${({ isOpen }) =>
    isOpen
      ? 0
      : "-30vw"}; // 창 크기 조절시 보이는 문제가 있어 의도적으로 멀리 두었습니다.
  top: 0;
  z-index: 999;
`;

const StyledMenuWrapper = styled.div`
  cursor: pointer;
  position: relative;
  height: 1.8rem;
  z-index: 1000;
`;

const Styledhamburger = styled.div<{ isOpen: boolean }>`
  position: relative;
  display: inline-block;
  left: 0;
  width: 1.8rem;
  height: 0.15rem;
  transition: all 0.8s;
  background-color: ${(props) =>
    props.isOpen
      ? css`
          ${({ theme }) => theme.color.glass}
        `
      : css`
          ${({ theme }) => theme.color.white};
        `};
  &::before,
  &::after {
    content: "";
    position: absolute;
    display: inline-block;
    left: 0;
    background-color: ${({ theme }) => theme.color.white};
    width: 1.8rem;
    height: 0.15rem;
    transition: all 0.8s;
  }
  &::before {
    top: ${(props) => (props.isOpen ? "0" : "-0.75rem")};
    transform: ${(props) =>
      props.isOpen ? "rotate(135deg)" : "rotate(0)"}; // 225
  }
  &::after {
    top: ${(props) => (props.isOpen ? "0" : "0.75rem")};
    transform: ${(props) => (props.isOpen ? "rotate(-135deg)" : "rotate(0)")};
  }
`;
