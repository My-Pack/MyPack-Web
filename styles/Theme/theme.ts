import { DefaultTheme } from "styled-components";

const color = {
  white: "#ffffff",
  black: "#000000",
  background: "#202124",
  buttonHoverBlack: "#3a3939",
  grey: "#353535",
  grey100: "#808080",
  grey500: "#6e6e6e",
  grey50: "#d3d3d3",
  borderBackground: "#2e2c2c",
  borderGrey: "#707070",
  glass: "rgba(0,0,0,0)", //투명
};

const fontWeight = {
  light: 300,
  normal: 500,
  semibold: 600,
  bold: 700,
};

const borderRadius = {
  input: "17px",
  button: "15px",
  uiCard: "10px",
};

export type ColorsTypes = typeof color;
export type FontWeightTypes = typeof fontWeight;
export type BorderRadiusTypes = typeof borderRadius;

const theme: DefaultTheme = {
  color,
  fontWeight,
  borderRadius,
};

export default theme;
