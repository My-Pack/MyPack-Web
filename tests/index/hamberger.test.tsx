import React from "react";
import { render } from "@testing-library/react";
import Hamburger from "src/components/Nav/Hamburger";
import { ThemeProvider } from "styled-components";
import theme from "styles/Theme/theme";

describe("Hamburger", () => {
  it("render btns", () => {
    const utils = render(
      <ThemeProvider theme={theme}>
        <Hamburger />
      </ThemeProvider>,
    );
    utils.getByText("내 정보");
    utils.getByText("알림");
    utils.getAllByText("나중에 추가");
  });
});
