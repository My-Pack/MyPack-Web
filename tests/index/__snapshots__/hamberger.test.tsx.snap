// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger snapshot test 1`] = `
<div>
  <div
    class="sc-75cb3926-2 eFkHLf"
  >
    <div
      class="sc-75cb3926-3 YhBch"
    />
  </div>
  <div
    class="sc-75cb3926-1 iVYOqf"
  >
    <p
      class="sc-75cb3926-0 jtANJk"
    >
      내 정보
    </p>
    <p
      class="sc-75cb3926-0 jtANJk"
    >
      알림
    </p>
    <p
      class="sc-75cb3926-0 jtANJk"
    >
      나중에 추가
    </p>
    <p
      class="sc-75cb3926-0 jtANJk"
    >
      나중에 추가
    </p>
  </div>
</div>
`;
